src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/Compression/ZLibStreamPool.cs

```diff
@@ -0,0 +1,283 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using static System.IO.Compression.ZLibNative;
+
+namespace System.Net.WebSockets.Compression
+{
+    internal sealed class ZLibStreamPool
+    {
+        private static readonly ZLibStreamPool?[] s_pools
+            = new ZLibStreamPool[WebSocketDeflateOptions.MaxWindowBits - WebSocketDeflateOptions.MinWindowBits + 1];
+
+        /// <summary>
+        /// The maximum number of cached items.
+        /// </summary>
+        private static readonly int MaximumRetained = Environment.ProcessorCount * 2;
```

Comment:
Not really. I guess even if a bad actor tries to ddos me with this, it would make much difference whether the streams are in the pool, or are in the websocket in the middle of receiving a message and the "bad" socket not providing further data and thus keeping the streams taken.

Let me know if you think this should be removed and the size of the pool should be controlled only by the timeout.